Directory.Build.targets

```diff
@@ -77,7 +77,15 @@
     <!-- Keep in sync as required by the Packaging SDK in Arcade. -->
     <Description>$(PackageDescription)</Description>
   </PropertyGroup>
-  
+
+  <!-- Make P2Ps which consume source generators, reference the netstandard2.0 TFM.
+       This allows source generators to multi-target without participating in the asset selection.
+       Uses the Microsoft.DotNet.Build.Task.TargetFramework infrastructure. -->
+  <ItemGroup>
+    <ProjectReference Update="@(ProjectReference->WithMetadataValue('OutputItemType', 'Analyzer')->WithMetadataValue('ReferenceOutputAssembly', 'false'))"
+                      ReferringTargetFramework="netstandard2.0" />
```

Comment:
It feels unusual to me that we'd ask our developers to set `OutputItemType="Analyzer" ReferenceOutputAssembly="false" ` and not set `ReferringTargetFramework="netstandard2.0"`.  All of these seem pretty equally obscure - so we either force them to understand the concepts and do it all (no magic in targets), or we help them with some simple convention (and they can trust us to get it right).   Feels like we're in the middle where we don't get either benefit fully.